cmds/mv/mv_test.go

```diff
@@ -43,32 +61,103 @@ func setup() (string, error) {
 	}
 
 	for _, t := range tests {
-		if err := ioutil.WriteFile(filepath.Join(d, t.n), []byte("hi"), t.m); err != nil {
+		if err := ioutil.WriteFile(filepath.Join(d, t.n), []byte(t.c), t.m); err != nil {
 			return "", err
 		}
 	}
 
 	return d, nil
 }
 
-func Test_mv_1(t *testing.T) {
+func TestMv(t *testing.T) {
 	d, err := setup()
 	if err != nil {
 		t.Fatal("err")
 	}
 	defer os.RemoveAll(d)
 
 	fmt.Println("Renaming file...")
-	files1 := []string{filepath.Join(d, "hi1.txt"), filepath.Join(d, "hi4.txt")}
-	if err := mv(files1, false); err != nil {
-		t.Error(err)
+	{
+		files := []string{filepath.Join(d, "hi1.txt"), filepath.Join(d, "hi4.txt")}
+		res := testutil.Command(t, files...)
+		_, err = res.CombinedOutput()
+		if err = testutil.IsExitCode(err, 0); err != nil {
+			t.Error(err)
+		}
 	}
 
 	dsub := filepath.Join(d, "hi.sub.dir")
 
 	fmt.Println("Moving files to directory...")
-	files2 := []string{filepath.Join(d, "hi2.txt"), filepath.Join(d, "hi4.txt"), dsub}
-	if err := mv(files2, true); err != nil {
+	{
+		files := []string{filepath.Join(d, "hi2.txt"), filepath.Join(d, "hi4.txt"), dsub}
+		res := testutil.Command(t, files...)
+		_, err = res.CombinedOutput()
+		if err = testutil.IsExitCode(err, 0); err != nil {
+			t.Error(err)
+		}
+	}
+}
+
+func TestMvUpdate(t *testing.T) {
+	*update = true
+	d, err := setup()
+	if err != nil {
+		t.Error(err)
+	}
+	defer os.RemoveAll(d)
+	fmt.Println("Testing mv -u...")
+
+	// Ensure that the newer file actually has a newer timestamp
+	currentTime := time.Now().Local()
+	oldTime := currentTime.Add(-10 * time.Second)
+	err = os.Chtimes(filepath.Join(d, old.n), oldTime, oldTime)
+	if err != nil {
 		t.Error(err)
 	}
+	err = os.Chtimes(filepath.Join(d, new.n), currentTime, currentTime)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Check that it doesn't downgrade files with -u switch
+	{
+		files := []string{"-u", filepath.Join(d, old.n), filepath.Join(d, new.n)}
+		res := testutil.Command(t, files...)
+		_, err = res.CombinedOutput()
+		if err = testutil.IsExitCode(err, 0); err != nil {
+			t.Error(err)
+		}
+		newContent, err := ioutil.ReadFile(filepath.Join(d, new.n))
+		if err != nil {
+			t.Error(err)
+		}
+		if bytes.Equal(newContent, old.c) {
+			t.Error("Newer file was overwritten by older file. Should not happen with -u.")
+		}
+	}
+
+	// Check that it does update files with -u switch
+	{
+		files := []string{"-u", filepath.Join(d, new.n), filepath.Join(d, old.n)}
+		res := testutil.Command(t, files...)
+		_, err = res.CombinedOutput()
+		if err = testutil.IsExitCode(err, 0); err != nil {
+			t.Error(err)
+		}
+		newContent, err := ioutil.ReadFile(filepath.Join(d, old.n))
+		if err != nil {
+			t.Error(err)
+		}
+		if !bytes.Equal(newContent, new.c) {
+			t.Error("Older file was not overwritten by newer file. Should happen with -u.")
+		}
+		if _, err := os.Lstat(filepath.Join(d, old.n)); err != nil {
+			t.Error("The new file shouldn't be there anymore.")
+		}
+	}
+}
```

Comment:
These are awesome, and thanks for improving the useless tests.

We should be checking that content & inode no are the same after the move. #949